test/playlist-selectors.test.js

```diff
@@ -51,3 +61,301 @@ test('Exponential moving average has a configurable decay parameter', function(a
   playlist = fiftyPercentDecay.call(hls);
   assert.equal(playlist.attributes.BANDWIDTH, 50, 'selected the middle playlist');
 });
+
+test('getBandwidth appropriately uses average bandwidth when specified',
+function(assert) {
+  let playlist = {};
+
+  assert.equal(getBandwidth(playlist, false),
+               void 0,
+               'undefined when playlist has no info');
+  assert.equal(getBandwidth(playlist, true),
+               void 0,
+               'undefined when playlist has no info');
+
+  playlist = {
+    attributes: {}
+  };
+
+  assert.equal(getBandwidth(playlist, false),
+               void 0,
+               'bandwidth 0 when playlist has no info');
+  assert.equal(getBandwidth(playlist, true),
+               void 0,
+               'bandwidth 0 when playlist has no info');
+
+  playlist = {
+    attributes: {
+      BANDWIDTH: 9
+    }
+  };
+
+  assert.equal(getBandwidth(playlist, false), 9, 'uses BANDWIDTH when available');
+  assert.equal(getBandwidth(playlist, true), 9, 'uses BANDWIDTH when only one available');
+
+  playlist.attributes['AVERAGE-BANDWIDTH'] = 3;
+
+  assert.equal(getBandwidth(playlist, false),
+               9,
+               'uses BANDWIDTH when useAverageBandwidth is false');
+  assert.equal(getBandwidth(playlist, true),
+               3,
+               'uses AVERAGE-BANDWIDTH when useAverageBandwidth is true');
+});
+
+test('crossedLowWaterLine detects when we have filled enough buffer', function(assert) {
+  assert.notOk(crossedLowWaterLine(0, videojs.createTimeRanges([])),
```

Comment:
For this test (and any others that use any of the constants in `Config`, I think it would be a good idea to override the config option with a set value for the test, and then restore is at the end. That way down the line if we decide that there is a better default value for that constant, we don't have to change all the tests that are written in a way that assume the old default value